onnxscript/irbuilder.py

```diff
@@ -344,35 +353,25 @@ def to_graph_proto(self, use_default_type: bool = True):
         graph, _ = self.to_graph_and_functions(use_default_type=use_default_type)
         return graph
 
-    def get_opset_import(self):
+    def get_opset_import(self) -> Dict[str, int]:
         func_opset_imports = {}
         for s in self.stmts:
-            if s.module.domain not in func_opset_imports:
-                func_opset_imports[s.module.domain] = s.module.version
-            elif func_opset_imports[s.module.domain] != s.module.version:
-                # TODO: this conflict is caused by assigning the default version to
-                # literal operators. Not to extend this PR too much,
-                # it needs to be fixed in another PR.
-                # raise RuntimeError(
-                #     ff"There is a version conflict in domain: {s.module.domain!r},\
-                #         with {self.name!r}.")
+            if s.callee.opset.domain not in func_opset_imports:
+                func_opset_imports[s.callee.opset.domain] = s.callee.opset.version
+            elif func_opset_imports[s.callee.opset.domain] != s.callee.opset.version:
                 warnings.warn(
-                    f"There is a version conflict in domain: {s.module.domain!r}, "
+                    f"There is a version conflict in domain: {s.callee.opset.domain!r}, "
                     f"with {self.name!r}.",
                     category=UserWarning,
                 )
         return func_opset_imports
 
-    def to_function_proto(self, domain):
-        """Converts a function into a *FunctionProto* after it is parsed
-        by the converter.
-
-        Warning:
-            About default values
+    def to_function_proto(self, domain: str) -> onnx.FunctionProto:
+        """Converts this instance into a `onnx.FunctionProto`.
 
-            Default values for attributes are introduced in onnx==1.13.0.
-            If an earlier version of onnx is installed, it ignores the default
-            values of the function arguments.
+        Note: Default values for attributes are an experimental feature in ONNX.
+        Conversion ignores default values for attributes if the ONNX version installed
+        doesn't support it.
         """
         opsets = self.get_opset_import()
         if domain != "":
```

Comment:
domian can be a type other string right

Reply:
I don't understand the question. The type is string. (But I don't know why we have this parameter, I think we should eventually eliminate it. I will look into it.)

Reply:
Oh, you are referring to the next line, I see now. Anyway, let me see if I can delete this parameter.